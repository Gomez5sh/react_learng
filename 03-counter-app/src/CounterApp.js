//Task
import React, { useState } from "react";
import PropTypes from "prop-types";

const CounterApp = ({ value }) => {
  //usestate
  const [counter, setCounter] = useState(value);

  //handles
  const count = (e) => {
    setCounter(counter + 1);
    //setCounter((c) => c + 1);
  };

  const minus = (e) => {
    setCounter(counter - 1);
  };

  const resetAll = (e) => {
    setCounter(value);
  };

  return (
    <>
      <h1>counterApp</h1>
      <h2>{counter}</h2>

      <button onClick={count}>+= 1</button>
      <button onClick={resetAll}>Reset</button>
      <button onClick={minus}>-= 1</button>
    </>
  );
};

CounterApp.prototypes = {
  valur: PropTypes.number.isRequired,
};

export { CounterApp as default };
