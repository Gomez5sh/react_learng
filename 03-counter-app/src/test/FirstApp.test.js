import React from "react";
import CounterApp from "../CounterApp";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
// import { render } from "@testing-library/react";

describe("Test on <CounterApp />", () => {
  /*test('Show message "Hi, i am a App" ', () => {
    const hi = "Hi, i am a App";
    const { getByText } = render(<FirstApp hello={hi} />);

    expect(getByText(hi)).toBeInTheDocument();
  });*/

  /*test("Show Firstapp ", () => {
    const hi = "Hi, i am a App";
    const wrapper = shallow(<FirstApp hello={hi} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("Show proops subtitle ", () => {
    const hi = "Hi, i am a App";
    const subtitle = "I am a subtitle";

    const wrapper = shallow(<FirstApp hello={hi} subtitle={subtitle} />);

    const PText = wrapper.find("p").text();
    expect(PText).toBe(subtitle);
  });*/

  let wrapper = shallow(<CounterApp value={10} />);

  beforeEach(() => {
    wrapper = shallow(<CounterApp value={10} />);
  });

  test("Show CounterApp ", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Show default value 100", () => {
    const wrapper = shallow(<CounterApp value={100} />);
    const newValue = wrapper.find("h2").text().trim();

    expect(newValue).toBe("100");
  });

  test("Increment value ", () => {
    wrapper.find("button").at(0).simulate("click");

    const newValue = wrapper.find("h2").text().trim();

    expect(newValue).toBe("11");
    console.log(newValue);
  });

  test("Drecementar value ", () => {
    wrapper.find("button").at(2).simulate("click");

    const newValue = wrapper.find("h2").text().trim();

    expect(newValue).toBe("9");
    console.log(newValue);
  });

  test("firts value ", () => {
    const wrapper = shallow(<CounterApp value={105} />);

    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(0).simulate("click");
    wrapper.find("button").at(1).simulate("click");
    const newValue = wrapper.find("h2").text().trim();
    console.log(newValue);
    expect(newValue).toBe("105");
  });
});
