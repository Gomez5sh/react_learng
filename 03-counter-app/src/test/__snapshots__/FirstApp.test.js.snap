// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <CounterApp /> Show CounterApp  1`] = `
<Fragment>
  <h1>
    counterApp
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    += 1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -= 1
  </button>
</Fragment>
`;
